### PR TITLE
Add event subscription support

### PR DESCRIPTION
**Pull request by nuustik**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -521,6 +547,16 @@ studio.internal = (function(proto) {
         app.makeGetterRequest(id, 1, 0, true);
       }
     }
+
+    this.async._makeEventRequest = function() {
+      if (eventSubscriptions.length > 0) {
+        var noZeroArray = eventSubscriptions.filter(v => v[1] != undefined);
+        var minTimeStamp = noZeroArray.length > 0 ? Math.min.apply(Math, noZeroArray.map(v => v[1])) : undefined;
```
Comment (Karmo7) [inline index.js:554]: I don't think Math.min is correct here. I would think each call to subscribeToEvents would give history based on the current startingFrom argument not the minimum of all subscribers.

Comment (nuustik) [reply index.js]: thought new protobuf call with new timestamp overrides the old subscription, and thus to serve the old subscriber we must use the min timestamp to cover for all subscribers

Comment (nuustik) [reply index.js]: but changed the code, as i got feedback that new protobuf call with new timestamp does not override it

**Review by martlaak — APPROVED** [review]

File: studioapi.proto.js [hunk 2]
```diff
@@ -233,5 +237,59 @@ message ValueRequest {
                                    // missing or zero means all samples must be provided
   extensions 100 to max;
 }
+
+/** CDP Event request message. */
+message EventRequest {
+  optional uint32 node_id = 1; // Target should forward events sent by this node ID (and its children)
+  optional uint64 starting_from = 2; // If > 0, target should re-forward history of past events starting from this timestamp
```
[the remark this answers is not in the document]
Comment (nuustik) [reply studioapi.proto.js]: changed

**Pull request merged**